File: src/classes/VoarFoguete.ts
import PadraoVoaveis from "../interfaces/PadraoVoaveis";

class VoarFoguete implements PadraoVoaveis {
  private velocidade: number

  constructor() {
    this.velocidade = 1000
  }

  public voar(): string {
    return `Voando como um foguete. Velocidade: ${this.getVelocidade()}`
  }

  public getVelocidade(): number {
      return this.velocidade
  }
}

export default VoarFoguete

File: EngSoft.Jogo.Patos/src/engsoft/jogo/patos/Voar_Foguete.java
package engsoft.jogo.patos;

// Classe que implementa o método voar e getVelocidade a partir da interface Padrao_Voaveis.java
public class Voar_Foguete implements Padrao_Voaveis{
	
	private double velocidade;
	
	public Voar_Foguete()
	{
		velocidade = 1000;	
	}
	
	public String voar() {
		return "Voando como um foguete. Velocidade: " + getVelocidade();		
	}

	public double getVelocidade() {
		return velocidade;
	}	
}
